application/src/main/java/org/opentripplanner/transit/service/TimetableRepositoryIndex.java

```diff
@@ -50,6 +50,8 @@ class TimetableRepositoryIndex {
   private final Multimap<Route, TripPattern> patternsForRoute = ArrayListMultimap.create();
   private final Multimap<StopLocation, TripPattern> patternsForStop = ArrayListMultimap.create();
 
+  private final Map<StopLocation, LocalDate> endOfServiceDateForStop = new HashMap<>();
+  private final Map<FeedScopedId, LocalDate> endOfServiceDateForService = new HashMap<>();
```

Comment:
So this is only needed to initialize `endOfServiceDateForStop`? It feels a bit wasteful to have this as a member variable.